src/javaRosa.js

```diff
@@ -1296,6 +1296,16 @@ define([
         }
     }
 
+    function insertOutputRef(vellum, target, path, mug, dateFormat) {
+        var output = getOutputRef(path, dateFormat),
+            form = vellum.data.core.form;
+        util.insertTextAtCursor(target, output, true);
+        if (mug) {
+            vellum.warnOnCircularReference('label', form, mug, path, 'output value');
+            warnOnNonOutputableValue(form, mug, path);
```

Comment:
I think this one could be reworked into `dnd_stop`, but that might be out of scope for this PR